### PR TITLE
Homework13: Aksana Batyrova

### PR DESCRIPTION
**Pull request by AksanaBatyrova**
HW 12 - OOP pt.2

### CODE REVIEW

File: homework13/task1_cards_deck.py [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+class CardsDeck:
+    """This class define cards deck
+    Arguments:
+
+    Methods:
+    create_deck, shuffle
+    """
+    new_deck: list[str] = []
+
+    def create_deck(self, cls):
```
Comment (Stml89) [inline homework13/task1_cards_deck.py:40]: `cls` is using for class methods https://medium.com/plain-simple-software/python-styling-cls-vs-self-ec141c28d6d4

```suggestion
    def create_deck(self, card):
```

Comment (Stml89) [inline homework13/task1_cards_deck.py:40]: It can be done in `__init__` method

File: homework13/task1_cards_deck.py [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+    suit_list = ['Hearts', 'Diamonds', 'Clubs', 'Spades']
+
+    def __init__(self, number=None, suit=None):
+        self.number = number
+        self.suit = suit
+
+    def __str__(self):
+        return f'{self.suit} {self.number}'
+
+    def get_by_number(self, number):
```
Comment (Stml89) [inline homework13/task1_cards_deck.py:24]: `get` method should be related to the `CardsDeck` class

File: homework13/task2_currency_converter.py [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+
+
+class Person():
+    """This class describes a customer of an exchange office"""
+    def __init__(self, currency, amount):
+        self.currency = currency
+        self.amount = amount
+
+    def __str__(self):
+        pass
```
Comment (Stml89) [inline homework13/task2_currency_converter.py:53]: if you introduce this method, so pls use it

File: homework13/task2_currency_converter.py [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+    Methods:
+    deposit(): calculating how much bank client will receive when depositing at
+    compound interest
+    """
+
+    def deposit(self, n, r):
+        """This method is for calculating compound interest"""
+        return int(n*(1+0.1)**r)
+
+    def exchange_currency(self, sell, amount, buy='BYN'):
```
Comment (Stml89) [inline homework13/task2_currency_converter.py:28]: move it to `Currency` class

**Review by Stml89 — CHANGES_REQUESTED** [review]

File: homework13/task1_cards_deck.py (hunk 1 above)
Comment (AksanaBatyrova) [reply homework13/task1_cards_deck.py]: fixed

Comment (AksanaBatyrova) [reply homework13/task1_cards_deck.py]: fixed

File: homework13/task1_cards_deck.py (hunk 2 above)
Comment (AksanaBatyrova) [reply homework13/task1_cards_deck.py]: fixed

File: homework13/task2_currency_converter.py (hunk 3 above)
Comment (AksanaBatyrova) [reply homework13/task2_currency_converter.py]: fixed

File: homework13/task2_currency_converter.py (hunk 4 above)
Comment (AksanaBatyrova) [reply homework13/task2_currency_converter.py]: fixed

File: homework13/task1_cards_deck.py [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+    create_deck, shuffle
+    """
+    new_deck: list[str] = []
+
+    # def create_deck(self, card_cls):
+    #     """This method is used to create cards deck"""
+    #     for s in card_cls.suit_list:
+    #         for n in card_cls.number_list:
+    #             self.new_deck.append(f'{s} {n}')
+    #     return self.new_deck
```
Comment (Stml89) [inline homework13/task1_cards_deck.py:39]: remove it

**Review by Stml89 — CHANGES_REQUESTED** [review]

File: homework13/task1_cards_deck.py (hunk 5 above)
Comment (AksanaBatyrova) [reply homework13/task1_cards_deck.py]: done

File: homework13/task1_cards_deck.py [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+    Arguments:
+
+    Methods:
+    create_deck, shuffle
+    """
+    new_deck: list[str] = []
+
+    def __init__(self):
+        self.new_deck = [
+            Card(number, suit)
```
Comment (Stml89) [inline homework13/task1_cards_deck.py:36]: will it create 4 Jokers? But, actually only 2 exists

File: homework13/task2_currency_converter.py [hunk 7]
```diff
@@ -0,0 +1,69 @@
...
+# класс Currency, который умеет конвертировать различные валюты(USD, EUR, ...)
+# в заданную валюту.
+
+# bank = Bank(..)
+
+# vasya = Person('USD', 10)
+# petya = Person('EUR', 5)
+
+
+class Bank:
```
Comment (Stml89) [inline homework13/task2_currency_converter.py:14]: Seems like you mossed functionality from previous HW(deposit)

**Review by Stml89 — CHANGES_REQUESTED** [review]

File: homework13/task1_cards_deck.py (hunk 6 above)
Comment (AksanaBatyrova) [reply homework13/task1_cards_deck.py:37]: fixed

File: homework13/task2_currency_converter.py (hunk 7 above)
Comment (AksanaBatyrova) [reply homework13/task2_currency_converter.py:14]: fixed

File: .github/workflows/ruff.yaml [hunk 8]
```diff
@@ -21,4 +21,4 @@ jobs:
           pip install ruff
       - name: Analysing the code with Ruff
         run: |
-          ruff --output-format=github .
+          ruff check --output-format=github .
```
Comment (Stml89) [inline .github/workflows/ruff.yaml:24]: What is the porpoise for adding this?

Comment (AksanaBatyrova) [reply .github/workflows/ruff.yaml:24]: image

**Review by Stml89 — APPROVED** [review]

**Pull request merged**